Wellthier/View Controllers/SearchViewController.m

```diff
@@ -9,147 +9,154 @@
+    [self setSearchBarPressed:NO];
     // Do any additional setup after loading the view.
-    [self resetButtonColors];
     self.tableView.dataSource = self;
+    self.tableView.indicatorStyle = UIScrollViewIndicatorStyleWhite;
+    self.tableView.keyboardDismissMode = UIScrollViewKeyboardDismissModeOnDrag;
     self.tableView.delegate = self;
+    self.collectionView.delegate = self;
+    self.collectionView.dataSource = self;
     self.searchBar.delegate = self;
     self.tableView.rowHeight = UITableViewAutomaticDimension;
-    [self getAllExercises];
 }
 
+
+
 - (void) getAllExercises {
-//    NSDictionary *headers = @{ @"X-RapidAPI-Key": @"59155f3b42msh4c970b09e3798d7p149cb3jsnd367e6d28891",
-//                               @"X-RapidAPI-Host": @"exercisedb.p.rapidapi.com" };
-//
-//    NSMutableURLRequest *request = [NSMutableURLRequest requestWithURL:[NSURL URLWithString:@"https://exercisedb.p.rapidapi.com/exercises"]
-//                                                           cachePolicy:NSURLRequestUseProtocolCachePolicy
-//                                                       timeoutInterval:10.0];
-//    [request setHTTPMethod:@"GET"];
-//    [request setAllHTTPHeaderFields:headers];
-//
-//    NSURLSession *session = [NSURLSession sharedSession];
-//    NSURLSessionDataTask *dataTask = [session dataTaskWithRequest:request
-//                                                completionHandler:^(NSData *data, NSURLResponse *response, NSError *error) {
-//                                                    if (error) {
-//                                                        NSLog(@"%@", error);
-//                                                    } else {
-//                                                        NSHTTPURLResponse *httpResponse = (NSHTTPURLResponse *) response;
-//                                                        //NSLog(@"%@", httpResponse);
-//                                                        NSDictionary *dataDictionary = [NSJSONSerialization JSONObjectWithData:data options:NSJSONReadingMutableContainers error:nil];
-//                                                        NSLog(@"%@", dataDictionary[@"results"]);
-////                                                        self.exercises = [Exercise exercisesWithDictionaries:dataDictionary[@"results"]];
-////                                                        NSLog(@"%@", self.exercises);
-//                                                    }
-//                                                }];
-//    [dataTask resume];
     ExerciseAPIManager *manager = [ExerciseAPIManager new];
     [manager fetchAllExercises:^(NSArray *exercises, NSError *error) {
         self.arrayOfExercises = (NSMutableArray *) exercises;
         self.filteredExercises = (NSMutableArray *) exercises;
-        //NSLog(exercises[0]);
-        
+        dispatch_async(dispatch_get_main_queue(), ^{
+            [self.tableView reloadData];
```

Comment:
You should create __weakSelf block variable outside block to ensure this doesn't crash.